rust/dyn-ddlog/src/ast/debug_printing.rs

```diff
@@ -0,0 +1,290 @@
+#![cfg(feature = "debug-printing")]
+
+use crate::ast::{
+    Attribute, Declaration, DeclarationKind, Expr, Fact, Function, Ident, IdentKind, Relation,
+    Rule, RuleClause, RuleHead,
+};
+use pretty::{Arena, DocAllocator, DocBuilder};
+use std::io::{self, Write};
+
+pub trait DebugAst {
```

Comment:
No, re-rendering source code is out of scope since [formatting is really hard](https://journal.stuffwithstuff.com/2015/09/08/the-hardest-program-ive-ever-written/)

Reply:
Pretty-printing is hard. Generating valid and semi-human readable code is straightforward and can help with debugging.